tokens/usecase/tokens_usecase.go

```diff
@@ -110,18 +122,12 @@ func NewTokensUsecase(tokenMetadataByChainDenom map[string]domain.Token) mvc.Tok
 		precisionScalingFactors[precision] = tenDec.Power(uint64(precision))
 	}
 
-	return &tokensUseCase{
-		tokenMetadataByChainDenom: tokenMetadataByChainDenom,
-		humanToChainDenomMap:      humanToChainDenomMap,
-		precisionScalingFactorMap: precisionScalingFactors,
-
-		pricingStrategyMap: map[domain.PricingSourceType]domain.PricingSource{},
-
-		poolDenomMetaData: sync.Map{},
+	t.tokenMetadataByChainDenom = tokenMetadataByChainDenom
+	t.humanToChainDenomMap = humanToChainDenomMap
+	t.precisionScalingFactorMap = precisionScalingFactors
 
-		chainDenoms:  chainDenoms,
-		coingeckoIds: coingeckoIds,
-	}
+	t.chainDenoms = chainDenoms
+	t.coingeckoIds = coingeckoIds
```

Comment:
I think we should make all of these `sync.Map`. Then, iterate over reach updated map and update it element-by-element.

In the current approach, we would get a concurrent map read and write.

In the alternative approach of surrounding these with locks, I'm afraid that we would get a tail latency increase due to copying entire maps under locks.

Reply:
Thanks, I have updated code to reflect your idea. :)